src/classes/ERR_ExceptionHandler.cls

```diff
@@ -100,11 +100,18 @@ public with sharing class ERR_ExceptionHandler {
 
     public static void handleAfflNullRecordTypeException(Affiliation__c affl, List<String> validRecordTypes) {
         if (affl.Account__c != null) {
+            // Get related Account's Record Type Name from 'Affiliation_Type__c' field
+            String afflAccRecTypeName;
+            if (affl.Affiliation_Type__c != null) {
+                AFFL_MultiRecordTypeMapper affilMapper = new AFFL_MultiRecordTypeMapper();
+                afflAccRecTypeName = affilMapper.accRecTypeLabelToAPI.get(affl.Affiliation_Type__c);
+            }
             //If parent account exists for affiliation and account has no record type
             //display the error message
-            if (affl.Affiliation_Type__c != null && !validRecordTypes.contains(affl.Affiliation_Type__c)) {
+            if ((affl.Affiliation_Type__c != null && !validRecordTypes.contains(affl.Affiliation_Type__c)) && 
```

Comment:
should this be API || Name instead?

Reply:
If the validRecordTypes does not contain both API Name of the Affiliation AND its Label Name then we throw an error. If either one is available in 'validRecordTypes' then an error is not thrown. Let me know if this clarifies your question @xinnanlisf .